include/zos-char-util.h

```diff
@@ -233,25 +233,25 @@ inline unsigned strlen_ae(const unsigned char *str, int *code_page,
   unsigned long code_out;
   const unsigned char *start;
 
-  bytes = max_len;
+  bytes = (unsigned long)max_len;
   code_out = 0;
   start = str;
   __asm volatile(" trte %1,%3,0\n"
                  " jo *-4\n"
                  : "+NR:r3"(bytes), "+NR:r2"(str), "+r"(bytes), "+r"(code_out)
                  : "NR:r1"(_tab_a)
                  :);
-  unsigned a_len = str - start;
+  unsigned a_len = (unsigned)((unsigned long)str - (unsigned long)start);
```

Comment:
Don't cast str & start to unsigned long.  The ptr arith will produce the correct difference.  Adding the casts will break.